scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -290,33 +301,44 @@ final case class GenericRecordParseIO[T](path: String, parseFn: GenericRecord =>
   /** Writes are undefined for [[GenericRecordParseIO]] since it is used only for reading. */
   override protected def write(data: SCollection[T], params: Nothing): Tap[T] = ???
 
-  override def tap(read: Unit): Tap[T] =
-    GenericRecordParseTap[T](ScioUtil.addPartSuffix(path), parseFn)
+  override def tap(read: ReadP): Tap[T] =
+    GenericRecordParseTap[T](path, parseFn, read)
 }
 
 object AvroIO {
+
+  object ReadParam {
+    val DefaultSuffix: String = null
+
+    private[scio] def apply(params: WriteParam): ReadParam =
+      new ReadParam(params.suffix)
+  }
+
+  final case class ReadParam private (
+    suffix: String = ReadParam.DefaultSuffix
+  )
+
   object WriteParam {
-    private[scio] val DefaultNumShards = 0
-    private[scio] val DefaultSuffix = ""
-    private[scio] val DefaultCodec: CodecFactory = CodecFactory.deflateCodec(6)
-    private[scio] val DefaultMetadata: Map[String, AnyRef] = Map.empty
-    private[scio] val DefaultShardNameTemplate: String = null
-    private[scio] val DefaultTempDirectory = null
-    private[scio] val DefaultFilenamePolicySupplier = null
+    val DefaultNumShards: Int = 0
+    val DefaultSuffix: String = ".avro"
+    val DefaultCodec: CodecFactory = CodecFactory.deflateCodec(6)
+    val DefaultMetadata: Map[String, AnyRef] = Map.empty
+    val DefaultFilenamePolicySupplier: FilenamePolicySupplier = null
+    val DefaultPrefix: String = null
+    val DefaultShardNameTemplate: String = null
+    val DefaultTempDirectory: String = null
   }
 
   final case class WriteParam private (
     numShards: Int = WriteParam.DefaultNumShards,
-    private val _suffix: String = WriteParam.DefaultSuffix,
+    suffix: String = WriteParam.DefaultSuffix,
     codec: CodecFactory = WriteParam.DefaultCodec,
     metadata: Map[String, AnyRef] = WriteParam.DefaultMetadata,
+    filenamePolicySupplier: FilenamePolicySupplier = WriteParam.DefaultFilenamePolicySupplier,
+    prefix: String = WriteParam.DefaultPrefix,
     shardNameTemplate: String = WriteParam.DefaultShardNameTemplate,
-    tempDirectory: String = WriteParam.DefaultTempDirectory,
-    filenamePolicySupplier: FilenamePolicySupplier = WriteParam.DefaultFilenamePolicySupplier
```

Comment:
@kellen Should I revert the param ordering in the `WriteParam` to minimize breaking changes ?

Reply:
No, I don't think this is important.